inference-engine/include/cpp/ie_memory_state.hpp

```diff
@@ -12,63 +12,57 @@
 
 #include <string>
 
-#include "ie_imemory_state.hpp"
+#include "ie_blob.h"
 #include "details/ie_exception_conversion.hpp"
 #include "details/ie_so_loader.h"
 
 namespace InferenceEngine {
 
+IE_SUPPRESS_DEPRECATED_START
+class IVariableState;
+IE_SUPPRESS_DEPRECATED_END
+
 /**
  * @brief C++ exception based error reporting wrapper of API class IVariableState
  */
-class VariableState {
-    IVariableState::Ptr actual = nullptr;
-    details::SharedObjectLoader::Ptr plugin = {};
+class INFERENCE_ENGINE_API_CLASS(VariableState) {
```

Comment:
Can this class be used without `IVariableState`? For now it looks like just a wrapper around it. In that case (wrapper), I think it should be marked as deprecated too.

Reply:
IVariableState will be moved to private folder while IVariableState will become a forward declaration.

So, we want to keep only C++ API in public API and move interfaces to `cpp_interfaces`